addon/serializers/application.js

```diff
@@ -86,9 +86,18 @@ export default DS.RESTSerializer.extend({
         // the links property so the adapter can async call the
         // relationship.
         // The adapter findHasMany has been overridden to make use of this.
-        if ( options.relation ) {
-          hash.links = {};
-          hash.links[key] = { type: relationship.type, key: key };
+        if(options.relation) {
+          // hash[key] contains the response of Parse.com: eg {__type: Relation, className: MyParseClassName}
+          // this is an object that make ember-data fail, as it expects nothing or an array ids that represent the records
+          hash[key] = [];
+
+          // ember-data expects the link to be a string
+          // The adapter findHasMany will parse it
+          if (!hash.links) {
+            hash.links = {};
+          }
+
+          hash.links[key] = JSON.stringify({typeKey: relationship.type.typeKey, key: key});
```

Comment:
These changes were from a pull request by @Kira2 so hopefully we can get her involved. Without running any tests I'm not sure of the difference between typeKey and type. @mixonic could you elaborate on that?